lib/open_project/scm/manageable_repository.rb

```diff
@@ -30,32 +30,47 @@
 module OpenProject
   module Scm
     module ManageableRepository
-      ##
-      # We let SCM vendor implementation define their own
-      # types (e.g., for differences in the management of
-      # local vs. remote repositories).
-      #
-      # But if they are manageable by OpenProject, they must
-      # expose this type through +available_types+.
-      MANAGED_TYPE = :managed
-      ##
-      # Reads from configuration whether new repositories of this kind
-      # may be managed from OpenProject.
-      def manageable?
-        !managed_root.nil?
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        ##
+        # We let SCM vendor implementation define their own
+        # types (e.g., for differences in the management of
+        # local vs. remote repositories).
+        #
+        # But if they are manageable by OpenProject, they must
+        # expose this type through +available_types+.
+        def managed_type
+          :managed
+        end
+
+        ##
+        # Reads from configuration whether new repositories of this kind
+        # may be managed from OpenProject.
+        def manageable?
+          ! (disabled_types.include?(managed_type) || managed_root.nil?)
```

Comment:
Looked furthre down into the code. `manageable?` Seems to be the main decision point in creating and removing repositories.

Returning false after managed repositories become unavailable makes sense for `CreateManagedRepositoryService` that should not allow me to create a managed repository if it is not permitted.

Where I see it more critical is in the repo controller that only deletes `manageable?` repositories.

~~That means after someone makes managed repositories unavailable for me and I decide to delete them, they will not disappear from disk, because they are not manageable. Even if the operator of my instance kept the directory for managed repositories configured.~~

Arg okay... I read the code further. I think [this line](https://github.com/opf/openproject/blob/38c49e04e1acf6c80f1c37ee76a0e4000b46b765/app/controllers/repositories_controller.rb#L112) should read `if @project.repository.managed?` instead of `if @project.repository.manageable?`, because that is what the `after_destroy` hook is checking for. And that is what makes semantically the most sense ^^

And now that I totally descoped discussion in this PR: Why is the `DeleteManagedRepositoryService` a struct and not a class? :D